operations/patch_util.go

```diff
@@ -621,17 +621,22 @@ Continue?`, uncommittedChangesFlag), true), nil
 // If no dir is provided, we use the current working directory.
 // The branch argument is used to determine where to generate the merge base from, and any extra
 // arguments supplied are passed directly in as additional args to git diff.
-func loadGitData(dir, branch, ref, commits string, format bool, extraArgs ...string) (*localDiff, error) {
-	// branch@{upstream} refers to the branch that the branch specified by branchname is set to
-	// build on top of. This allows automatically detecting a branch based on the correct remote,
-	// if the user's repo is a fork, for example. This also works with a commit hash, if given.
+func loadGitData(dir, remote, branch, ref, commits string, format bool, extraArgs ...string) (*localDiff, error) {
+	if remote == "" {
+		remote = "upstream"
+	}
+	// remote/branch refers directly to the remote branch and does not require a local branch.
+	// branch@{remote} referes to a local branch that has a tag of a remote branch attached to it.
 	// In the case a range is passed, we only need one commit to determine the base, so we use the first commit.
 	// For details see: https://git-scm.com/docs/gitrevisions
 
-	mergeBase, err := gitMergeBase(dir, branch+"@{upstream}", ref, commits)
+	mergeBase, err := gitMergeBase(dir, remote+"/"+branch, ref, commits)
 	if err != nil {
-		return nil, errors.Wrapf(err, "Error getting merge base, "+
-			"may need to create local branch '%s' and have it track upstream", branch)
+		mergeBase, err = gitMergeBase(dir, branch+"@{"+remote+"}", ref, commits)
```

Comment:
re above: I think this still has to be `branch@{upstream}` because of the `@{upstream}` thing I mentioned above.